build-ps/debian/extra/mysql-systemd

```diff
@@ -0,0 +1,374 @@
+    esac
+
+    while /bin/true ; do
+        sleep 1
+        mysqladmin --no-defaults --socket="$adminsocket" --user=UNKNOWN_MYSQL_USER ping >/dev/null 2>&1 && break
+    done
+    exit 0
+}
+
+
+action=$1
+manager=${2:-0}
+service_startup_timeout=900
+startup_sleep=1
+basedir=/usr
+bindir="$basedir/bin"
+sbindir="$basedir/sbin"
+libexecdir="$basedir/libexec"
+
+env_args="${EXTRA_ARGS:-}"
+
+
+PATH="/sbin:/usr/sbin:/bin:/usr/bin:$bindir:$sbindir"
+export PATH
+
+pid_path=$(parse_cnf pid-file "" server mysqld mysqld_safe)
+datadir=$(parse_cnf datadir "" server mysqld mysqld_safe)
+service_startup_timeout=$(parse_cnf service-startup-timeout $service_startup_timeout mysqld_safe)
+
+if [[ -z ${datadir:-} ]];then
+    datadir="/var/lib/mysql"
+    if [[ ! -d $datadir ]];then
+        echo "FATAL: datadir not found in cnf or is /var/lib/mysql"
+        exit 2
+    fi
+fi
+
+grastate_loc="${datadir}/grastate.dat"
+
+if [[ -z "${pid_path:-}" ]];then
+  pid_path=$datadir/`hostname`.pid
+else
+  case "$pid_path" in
+    /* ) ;;
+    * )  pid_path="$datadir/$pid_path" ;;
+  esac
+fi
+
+if [[ $action == 'reload' || $action == 'stop' || $action == 'stop-post' ]];then
+    if [[ ! -s $pid_path ]];then
+        log_warning_msg "mysql pid file $pid_path empty or not readable"
+        [[ $action == 'stop' ]] && log_failure_msg "mysql already dead"
+        if [[ $action == 'stop-post' ]];then
+            log_warning_msg "mysql may be already dead"
+            exit
+        fi
+        exit 2
+    fi
+    mysql_pid=$(cat $pid_path)
+    if [[ $action == 'stop-post' && -n ${mysql_pid:-} ]]  && ! kill -0 $mysql_pid;then
+        log_warning_msg "mysql already dead"
+        log_failure_msg "Stale PID file: $pid_path"
+        exit 3
+    fi
+fi
+
+if [[ $action == 'start-post' ]];then
+    if ! kill -0 $manager;then
+        log_failure_msg "mysqld_safe with PID $manager has already exited"
+        exit 1
+    fi
+fi
+
+check_running()
+{
+    if [[ -s $pid_path ]];then
+	    mysql_pid=$(cat $pid_path)
+	    if [[ -n ${mysql_pid:-} ]]  && kill -0 $mysql_pid 2>/dev/null;then
+                mysql_existed_pid=$(ps wwaux | grep mysql | grep "wsrep-provider" | awk '{print $2}')
+                if [[ ${mysql_pid} == ${mysql_existed_pid} ]]; then
+                    log_warning_msg "PXC is in bootstrap mode.  To switch to normal operation, first stop the mysql@bootstrap.service then start the mysql service"
+                else
+                    log_warning_msg "Another instance of mysqld running on $mysql_pid, exiting.."
+                fi
+                exit 1
+
+	    fi
+    fi
+}
+
+# Run mysqld with --wsrep-recover and parse recovered position from log.
+# Position will be stored in wsrep_start_position_opt global.
+wsrep_start_position_opt=""
+wsrep_recover_position() {
+  local mysqld_cmd="/usr/sbin/mysqld"
+  local ret=0
+  local uuid=""
+  local seqno=0
+  local wr_logfile=$(mktemp $datadir/wsrep_recovery.XXXXXX)
```

Comment:
maybe wsrep_logfile would be better

Reply:
Done with adjustments requested by Ramesh in part of not including the log of "mysqld --wsrep-recover ..." execution in case of successful wsrep position recovery